goldilocks/src/lib.rs

```diff
@@ -11,14 +11,17 @@ use core::iter::{Product, Sum};
 use core::ops::{Add, AddAssign, Div, Mul, MulAssign, Neg, Sub, SubAssign};
 
 use p3_field::{
-    exp_10540996611094048183, exp_u64_by_squaring, AbstractField, Field, Packable, PrimeField,
-    PrimeField64, TwoAdicField,
+    exp_10540996611094048183, exp_u64_by_squaring, halve_u64, AbstractField, Field, Packable,
+    PrimeField, PrimeField64, TwoAdicField,
 };
 use p3_util::{assume, branch_hint};
 use rand::distributions::{Distribution, Standard};
 use rand::Rng;
 use serde::{Deserialize, Serialize};
 
+/// The Goldilocks prime
+const P: u64 = 0xFFFF_FFFF_0000_0001;
```

Comment:
nit: could also use `Goldilocks::ORDER_U64`, or if you prefer a separate global const, maybe change one to reference the other

Reply:
Yeah I think I've already done that?

I defined Goldilocks::ORDER_U64 as P.

The advantage of introducing P here is that it makes the Baby Bear, Mersenne and Goldilocks definitions more identical.

Reply:
Oh whoops! Yeah looks good